examples/cfq/input_pipeline.py

```diff
@@ -144,10 +144,12 @@ def get_example_length(self, example: ExampleType) -> tf.Tensor:
       example_len = tf.math.maximum(question_len, query_len)
     return example_len
 
-  def indices_to_sequence_string(self, indices: jnp.ndarray) -> Text:
+  def indices_to_sequence_string(self,
+                                 indices: jnp.ndarray,
+                                 keep_padding: bool = True) -> Text:
     """Transforms a list of vocab indices into a string
         (e.g. from token indices to question/query)"""
-    tokens = [self.i2w[i].decode() for i in indices]
+    tokens = [self.i2w[i].decode() for i in indices if keep_padding or i!=0]
```

Comment:
I don't really understand how this works, what padding will it remove? Currently, it seems that if `keep_padding == False`, then this will output the entire sequence except the first index. Don't you want to remove padding tokens later in the sequence as well?

Reply:
The padding token has index zero. This convention is used throughout the code (also when the padding is done). What this piece of code does is filter out the zero-valued indices, the indices of padding tokens, when keep_padding = False. I've explained this in the function description now.

Reply:
Ohh right, i is the index. Sorry I was confused. It makes sense now, thanks for explaning!